app/src/main/java/com/fdiazblockone/onboarding/presentation/detail/DetailViewModel.kt

```diff
@@ -0,0 +1,25 @@
+package com.fdiazblockone.onboarding.presentation.detail
+
+import androidx.lifecycle.ViewModel
+import com.fdiazblockone.onboarding.interactors.GetBlockById
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class DetailViewModel(private val blockId: String, private val getBlockById: GetBlockById) :
+    ViewModel() {
+
+    private var viewModelJob = Job()
+    private val coroutineScope = CoroutineScope(viewModelJob + Dispatchers.Main)
+
+    init {
+        getBlockById()
+    }
+
+    private fun getBlockById() {
+        coroutineScope.launch {
```

Comment:
Since you are using an Android ViewModel, you can save yourself the `Job` and the `CoroutineScope` and use the viewModel's `viewModelScope` directly